### ISSUE TITLE
[Feature request] filtering runs by summary metrics

### ISSUE THREAD

**Issue by maria-korosteleva** [wandb/client#1369]
It would be great to extend the filtering in the run table\dashboard and allow summary metrics to be keys in filters.

Use-case: 
I'm running a sweep, and want to take a closer look at the runs that perform better than a certain loss threshold.  I don't see any easy way to filter those runs only right now. (If such a way exists, my apologies, please, point it out to me!).

**Comment by vanpelt** [comment]
Hey @maria-korosteleva summary metrics should be available in the table filters.  After clicking the "filter" icon you can choose either hyper-parameters or summary-metrics as filters as long as the values are scalars, dates, or strings:

![image](https://user-images.githubusercontent.com/17/96281827-bd432900-0f8e-11eb-8c36-ef8a7d886281.png)

If you have a workspace where this doesn't seem to be working, send a link my way and I'll look into what's going on.

**Comment by maria-korosteleva** [comment]
Hi @vanpelt, thank you for your reply!
Indeed, I found summary metrics in filters! They were not showing in the initial list of available keys but started to show up when I  started typing something in the field. This solves my problem!
But should not summary metrics be showing up in the initial list as well? 

Here is the dashboard I'm working with
https://wandb.ai/maria_korosteleva/Garments-Reconstruction/sweeps/deoi45kc?workspace=user-maria_korosteleva

**Comment by vanpelt** [comment]
Hmm, going to file a ticket for this one, we should have an infinite scroll that would make this show up in the initial list.

**Comment by maria-korosteleva** [comment]
Double-checked today. 
It seems to have become worse: the summary metrics neither appear in the initial list now when I try to search for them explicitly! 😢 

Old filters based on the summary metrics seem to be working
![image](https://user-images.githubusercontent.com/6837635/102735975-d1235680-4354-11eb-97c1-7e53433052a7.png)

**Comment by ariG23498** [comment]
Hey @maria-korosteleva 
If I am able to get your issue correctly, the metrics of the sweep is neither showing in the initial list nor when you type it yourself in the text field.
Could you check whether the metrics that you are looking for, are present in the `overview` page of the sweep?

If you seem to find the metrics, could you send our way a minimal implementation of this ticket?
Thanks in advance. :smile:

**Comment by maria-korosteleva** [comment]
Hi @ariG23498 
As of today, summary metrics do not show up in **any filters**, neither for sweeps nor for the project main dashboard (the screenshot in my previous message is from the project dashboard).

I have the impression that this problem is on the web-UI side, so could you explain what do you want to see as "a minimal implementation"?

To give you a more detailed example, here is the sweep, https://wandb.ai/maria_korosteleva/Garments-Reconstruction/sweeps/6p7vpvpz?workspace=user-maria_korosteleva

As you can see from the screenshot below, I have old filters set up to use the `best_validation_loss` summary metric, but when I try to create a new filter with this metric, it's not able to find anything. 
This is an old sweep, and yes, I'm 100% sure `best_validation_loss` was properly logged -- this is the target metric in the sweep. The **old filters with `best_validation_loss`  were working just fine at the time I was working with this sweep**. 

![image](https://user-images.githubusercontent.com/6837635/102839568-e316ff00-4411-11eb-95c2-67bb3458a09d.png)

**Comment by ariG23498** [comment]
Hey @maria-korosteleva 
I ran a sweep myself. I have also taken a snap of the hyperprams and the summary metrics that I sweep over.
![image](https://user-images.githubusercontent.com/36856589/102847510-83801a00-4438-11eb-8d12-2646203a6742.png)

In the filter section I indeed am able to view the hyperparams and the summary metrics.
![image](https://user-images.githubusercontent.com/36856589/102847401-46b42300-4438-11eb-8faf-e6e13dcd1096.png)


If I am missing anything please let me know. :smile:

**Comment by vanpelt** [comment]
@maria-korosteleva we just deployed a fix to the frontend that was preventing some of the values from displaying.  I've confirmed in your sweep that `best_valid_loss` now displays in new filters.  Sorry about that.

**Comment by maria-korosteleva** [comment]
Yep, I'm able to find my summary metrics in the filters now! Thank you, @vanpelt!!


> Hmm, going to file a ticket for this one, we should have an infinite scroll that would make this show up in the initial list.

I don't think the infinite scroll is there though =( I still don't see summary metrics in the initial list of properties I can filter by (before I start searching).
